src/ar_poller.erl

```diff
@@ -1,84 +1,99 @@
-				unavailable ->
-					server(Node, Peers, LastBH)
-			end
-	end.
+%%% This module fetches blocks from trusted peers in case the node is not in the
+%%% public network or hasn't received blocks for some other reason.
 
-get_remote_block_hash_list([]) ->
-	unavailable;
-get_remote_block_hash_list([Peer | Peers]) ->
-	case catch ar_http_iface_client:get_hash_list(Peer) of
-		{'EXIT', _} ->
-			get_remote_block_hash_list(Peers);
-		BHL ->
-			{BHL, Peer}
-	end.
+%% The polling frequency in seconds.
+-define(DEFAULT_POLLING_INTERVAL, 60 * 1000).
 
-get_remote_block_pair(Peer, BHL) ->
-	case ar_node_utils:get_full_block(Peer, hd(BHL), BHL) of
-		unavailable -> unavailable;
-		B -> get_remote_block_pair(Peer, BHL, B)
-	end.
+%%%===================================================================
+%%% Public API.
+%%%===================================================================
 
-get_remote_block_pair(Peer, BHL, B) ->
-	%% Fetch the previous block, since that's required before calling
-	%% ar_util:get_recall_hash/2
-	case get_block(B#block.previous_block, BHL, Peer) of
-		{ok, _} ->
-			PreviousRecallBH = ar_util:get_recall_hash(B#block.previous_block, BHL),
-			case get_block(PreviousRecallBH, BHL, Peer) of
-				{ok, PreviouRecallB} -> {B, PreviouRecallB};
-				unavailable -> unavailable
-			end;
-		unavailable ->
-			unavailable
-	end.
+start_link(Args) ->
+	gen_server:start_link({local, ?MODULE}, ?MODULE, Args, []).
+
+set_current_height(Height) ->
+	gen_server:cast(?MODULE, {set_height, Height}).
 
-get_block(BH, BHL, Peer) ->
-	case ar_storage:read_block(BH, BHL) of
+%%%===================================================================
+%%% Generic server callbacks.
+%%%===================================================================
+
+init(Args) ->
+	ar:info([{event, ar_poller_start}]),
+	I = proplists:get_value(polling_interval, Args, ?DEFAULT_POLLING_INTERVAL),
+	start_polling(),
+	{ok, #{
+		trusted_peers => proplists:get_value(trusted_peers, Args, []),
+		height => -1,
+		interval => I,
+		next_call_timer => not_set
+	}}.
+
+handle_cast(schedule_polling, #{ height := Height, interval := Interval} = State) ->
+	case Height of
+		-1 ->
+			%% Wait until the node joins the network or starts from a hash list.
+			do_nothing;
+		_ ->
+			poll_block(maps:get(trusted_peers, State, []), Height + 1),
+			ok
+	end,
+	erlang:send_after(Interval, ?MODULE, schedule_polling),
+	{noreply, State};
+handle_cast({set_height, Height}, State) ->
+	#{ next_poll_timer := NextPollTimer, interval := Interval } = State,
+	cancel_timer(NextPollTimer),
+	erlang:send_after(Interval, ?MODULE, schedule_polling),
+	{noreply, State#{ height => Height }}.
+
+handle_call(_Request, _From, State) ->
+	{noreply, State}.
+
+%%%===================================================================
+%%% Internal functions.
+%%%===================================================================
+
+start_polling() ->
+	gen_server:cast(?MODULE, schedule_polling).
```

Comment:
So I would do `timer:apply_interval(Interval, gen_server, cast, [?MODULE, schedule_polling])` here and store the returned timer reference in the gen_server's state. It could then be canceled using that `cancel_timer` function using that reference